addons/project/models/project.py

```diff
@@ -1333,6 +1344,16 @@ def _compute_dependent_tasks_count(self):
             for task in tasks_with_dependency:
                 task.dependent_tasks_count = dependent_tasks_count_dict.get(task.id, 0)
 
+    @api.depends('depend_on_ids', 'depend_on_ids.stage_id', 'depend_on_ids.stage_id.fold')
+    def _is_blocked(self):
+        for task in self:
+            task.is_blocked = any(not blocking_task.stage_id.fold for blocking_task in task.depend_on_ids)
+
+    @api.depends('depend_on_ids', 'dependent_ids', 'stage_id', 'stage_id.fold')
+    def _is_blocking(self):
+        for task in self:
+            task.is_blocking = not task.stage_id.fold and task.dependent_ids
```

Comment:
To avoid fetching all the dependent tasks in self we could use the `dependent_tasks_count` to check if each task depend on at least another task. 🙂 

I have added the search method too
```suggestion
    @api.depends('dependent_tasks_count', 'is_closed')
    def _compute_is_blocking(self):
        for task in self:
            task.is_blocking = not task.is_closed and task.dependent_tasks_count > 0
            
    def _search_is_blocking(self, operator, value):
        if operator not in ['=', '!=']:
            raise NotImplemented(f'The operator {operator} is not supported in this search method.')
        domain = ['&', ('is_closed', '=', False), ('dependent_ids', '!=', [])]
        if operator == '!=' and value is True or operator == '=' and value is False:
            domain.insert(0, expression.NOT_OPERATOR)
        domain = expression.distribute_not(domain)
        return domain
```